### PR TITLE
Copymitter Fix

### PR DESCRIPTION
**Pull request by ericfranz**
This adds the tests below. Separate PRs to merge into this add the actual fix.

### CODE REVIEW

File: lib/cloudcmd/lib/server/rest.js [hunk 1]
```diff
@@ -249,124 +249,128 @@
...
         cp.on('end', function() {
             process.stdout.write('\n');
             fn(error);
         });
-    }
-    
+    };
+
+    // expose copy function for testing purposes
+    module.exports.copy = copy;
```
Comment (MorganRodgers) [inline lib/cloudcmd/lib/server/rest.js]: Be careful with unnecessary whitespace changes. There are so many in this file that it is difficult to see where the actual code changes are.

File: test/copy.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
+var assert = require('assert'),
+    rest = require('cloudcmd/lib/server/rest'),
+    path = require('path'),
+    fs   = require('fs-extra'), // use fs-extra for removeSync
+    home = process.env.HOME,
+    scratch = process.env.SCRATCH,
+    fixtures_path = path.resolve(__dirname, 'fixtures/'),
+    hometmp, scratchtmp;
+
+function s(p){
```
Comment (MorganRodgers) [inline test/copy.js]: We shouldn't use one character function names or variable names outside of the conventional (`n`, `i` when writing c-style loop counters, and when doing cartesian coordinates `x`, `y`, `z`).

Comment (ericfranz) [reply test/copy.js]: In this one case I have mixed feelings because the single letter dramatically increases the readability of the code once you know what the single letter does. Resorting to just ensuring that a slash is appended to all strings is relying on state which in practices is prone to error. So in the end it's a set of bad tradeoffs.

Though no one is ever going to edit this again so...

Comment (ericfranz) [reply test/copy.js]: For example:

```js
rest.copy(s(fixtures_path), s(scratchtmp), ["data"], function(){
  rest.copy(s(scratchtmp), s(target), ["data"], function(){

// vs

rest.copy(add_slash(fixtures_path), add_slash(scratchtmp), ["data"], function(){
  rest.copy(add_slash(scratchtmp), add_slash(target), ["data"], function(){
```

Comment (ericfranz) [reply test/copy.js]: Well this is actually better:

```js
rest.copy(normalize(fixtures_path), normalize(scratchtmp), ["data"], function(){
  rest.copy(normalize(scratchtmp), normalize(target), ["data"], function(){
```

though `normalize` here != `path.normalize`
